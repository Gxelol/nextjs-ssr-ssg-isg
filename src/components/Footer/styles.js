import styled, { css } from 'styled-components';

import { Container as Text } from '../Text/styles';
import { Container as Section } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.primaryColor};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Section} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
