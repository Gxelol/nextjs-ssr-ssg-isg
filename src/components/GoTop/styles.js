import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    opacity: 0.8;
  `}
`;
