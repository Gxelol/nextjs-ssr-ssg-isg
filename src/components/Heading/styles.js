import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
    `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.big};
    `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.huge};
    ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.big};
  }
`;

const titleCase = (uppercase) => css`
text-transform: ${uppercase ? 'uppercase' : 'none'}
`;

export const Title = styled.h1`
  ${({ theme, darkColor, size, uppercase }) => css`
    color: ${darkColor ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
