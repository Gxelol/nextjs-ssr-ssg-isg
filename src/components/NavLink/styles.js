import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    text-transform: uppercase;

    @media ${theme.media.lteSmall} {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
