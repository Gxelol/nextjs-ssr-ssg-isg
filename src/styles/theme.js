export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#ffffff',
    lightGray: '#dddddd',
  },
  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      small: '1.6rem',
      minor: '2.4rem',
      medium: '3rem',
      big: '4.6rem',
      huge: '5.8rem',
    },
  },
  media: {
    lteSmall: '(max-width: 38.4rem)',
    lteMedium: '(max-width: 76.8rem) and (min-width: 38.5rem)',
  },
  spacings: {
    xsmall: '0.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
