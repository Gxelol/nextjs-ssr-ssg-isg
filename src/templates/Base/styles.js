import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 3rem;

    @media ${theme.media.lteSmall} {
      padding-top: 0;
    }
  `}
`;
