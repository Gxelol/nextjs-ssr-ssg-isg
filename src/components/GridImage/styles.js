import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
