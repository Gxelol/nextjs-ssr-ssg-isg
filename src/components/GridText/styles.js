import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Text } from '../Text/styles';

export const Container = styled.section`
  ${({ theme }) => css`
    ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${Heading} {
    position: relative;
    left: 5rem
  }

  ${Heading}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
  `}
`;
