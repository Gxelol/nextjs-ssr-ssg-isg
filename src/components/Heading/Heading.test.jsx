import { screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Heading } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  it('should render with default values', () => {
    renderTheme(<Heading> children </Heading>);
    const heading = screen.getByRole('heading', { name: 'children' });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.fonts.sizes.huge,
      'text-transform': 'none',
    });
  });

  it('should match snapshot', () => {
    renderTheme(<Heading> children </Heading>);
    const { container } = screen.getByRole('heading', { name: 'children' });
    expect(container).toMatchSnapshot();
  });

  it('should render with whiteColor', () => {
    renderTheme(<Heading darkColor={false}> children </Heading>);
    const heading = screen.getByRole('heading', { name: 'children' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render correct heading sizes', () => {
    const { rerender } = renderTheme(
      <Heading size="small"> children </Heading>,
    );
    const heading = screen.getByRole('heading', { name: 'children' });
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.small,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="medium"> children </Heading>
      </ThemeProvider>,
    );
    expect(screen.getByRole('heading', { name: 'children' })).toHaveStyle({
      'font-size': theme.fonts.sizes.medium,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="big"> children </Heading>
      </ThemeProvider>,
    );
    expect(screen.getByRole('heading', { name: 'children' })).toHaveStyle({
      'font-size': theme.fonts.sizes.big,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="huge"> children </Heading>
      </ThemeProvider>,
    );
    expect(screen.getByRole('heading', { name: 'children' })).toHaveStyle({
      'font-size': theme.fonts.sizes.huge,
    });
  });

  it('should render correct font-size when the screen is less than 76.8rem', () => {
    renderTheme(<Heading size="huge"> children </Heading>);
    screen.getByRole('heading', { name: 'children' });

    expect(screen.getByRole('heading', { name: 'children' })).toHaveStyleRule(
      'font-size',
      theme.fonts.sizes.big,
      {
        media: theme.media.lteMedium,
      },
    );
  });

  it('should render with uppercase', () => {
    renderTheme(<Heading uppercase> children </Heading>);
    const heading = screen.getByRole('heading', { name: 'children' });

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render correct heading element', () => {
    const { container } = renderTheme(<Heading as="h6"> children </Heading>);
    const h6 = container.firstChild;

    expect(h6.tagName.toLowerCase()).toBe('h6');
  });
});
