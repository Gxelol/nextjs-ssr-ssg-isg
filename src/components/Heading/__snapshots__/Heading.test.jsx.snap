// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `undefined`;
